api/stream/client.go

```diff
@@ -87,13 +90,22 @@ func (c *client) startWriteHandler(pingPeriod time.Duration) {
 
 	for {
 		select {
-		case message, ok := <-c.write:
+		case event, ok := <-c.write:
```

Comment:
Just a thought, maybe the event could be
```
type WebSocketEvent interface {
    ToExternal(apiVersion int) (interface{}, error)
}
```
With this, we can basically omit the event encoder and just let our events handle how v1 and v2 is structured.

Reply:
So the encoders have different ideas of how an event is to be encoded. Since I wanted all WebSocket messages to be reflected certain structs in the code, all event encoders maintain different structs for the events. I had this previously implemented as you said, but noticed it would be a lot cleaner to have a file per encoder, so the structs (which represent the JSON encoding) could be separated. I felt it made more sense this way, since the files separate the "ideas" on how to encode events.
Also, once a "version" is no longer needed (assuming clients will adopt the new version), we can simple remove one file and remove the mapping to it.

Reply:
Can we close this conversation?